cpu/stm32_common/include/periph_cpu_common.h

```diff
@@ -198,15 +198,25 @@ typedef struct {
     uint8_t chan;           /**< DAC device used for this line */
 } dac_conf_t;
 
+
+/**
+ * @brief   Timer type
+ */
+typedef enum {
+    TIMER_TYPE_GPTIM, /**< General-purpose timer */
+    TIMER_TYPE_LPTIM, /**< Low-power timer */
+} timer_type_t;
+
 /**
  * @brief   Timer configuration
  */
 typedef struct {
-    TIM_TypeDef *dev;       /**< timer device */
+    TIM_TypeDef *dev;       /**< timer device, not set for LPTIM */
```

Comment:
why not? Set it to `LPTIM1` and you can use `dev(tim)->` below in line 223, too.